scripts/generate-abis.ts

```diff
@@ -0,0 +1,68 @@
+import * as path from 'path';
+import * as fs from 'fs';
+
+/**
+ * This generates const TypeScript ABIs for each asset in
+ * `@safe-global/safe-deployments` package for `viem` to infer.
+ *
+ * Although it is possible to get a singleton programmatically and
+ * import the JSON directly, neither is strictly typed.
+ *
+ * Once it is possible to import JSON "as const", the deployments
+ * package should be updated to return the singletons as such.
+ *
+ * @see https://github.com/microsoft/TypeScript/issues/32063
+ */
+
+// Path to directory containing JSON assets
+const assetsDir = path.join(
+  process.cwd(),
+  'node_modules',
+  '@safe-global',
+  'safe-deployments',
+  'dist',
+  'assets',
+);
+
+// Path to directory where ABIs will be written
+const outputDir = path.join(process.cwd(), 'src', 'abis', 'safe');
```

Comment:
I don't think we should generate source files to the `src` folder as these are usually manually written and we should avoid mixing generated sources with the sources that we track on the version control.

This also creates the wrong expectation that if someone creates source code on `/src/abis` that source code would be deleted by this script but also ignored by Git (given the `.gitignore` rules).

Have you considered the `dist` folder of the project instead?

Reply:
It outputs to `/dist/abis/safe` as of 9dff79a72bbe821c2a4bb3861949c01770d340c4. I had to make some config adjustments to allow for `@/dist` imports but it should be self-explanatory.